app/Http/Middleware/LastActivity.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Auth;
+use Carbon\Carbon;
+
+class LastActivity
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        foreach(array_keys(config('auth.guards')) as $guard) {
+
+            $user = Auth::guard($guard)->user();
+
+            if ($user) {
+                $this->updateLastActivityField($user);
+            }
+        }
+
+        return $next($request);
+    }
+
+    /**
+     * Updates the last activity field for a specified model.
+     *
+     * @param Model $user
+     */
+    protected function updateLastActivityField(Model $user)
+    {
+        $lastActivityField = 'last_activity_at';
+        $graceTime = 5;
+
+        if ($graceTime > 0 && $user->$lastActivityField && (new Carbon($user->$lastActivityField))->addSeconds($graceTime) > now()) {
+            return;
+        }
```

Comment:
add options (example 5 seconds after user online, column 'last_activity_at' will be updated)